packages/framework-provider-local/src/library/graphql-adapter.ts

```diff
@@ -1,45 +1,116 @@
-import { BoosterConfig, GraphQLRequestEnvelope, GraphQLRequestEnvelopeError, UUID } from '@boostercloud/framework-types'
+import {
+  BoosterConfig,
+  EventType,
+  GraphQLClientMessage,
+  GraphQLOperation,
+  GraphQLRequestEnvelope,
+  GraphQLRequestEnvelopeError,
+  UUID,
+} from '@boostercloud/framework-types'
 import { getLogger } from '@boostercloud/framework-common-helpers'
 import * as express from 'express'
+import { ExpressWebSocketMessage } from './web-socket-server-adapter'
 
 export async function rawGraphQLRequestToEnvelope(
   config: BoosterConfig,
-  request: express.Request
+  request: express.Request | ExpressWebSocketMessage
 ): Promise<GraphQLRequestEnvelope | GraphQLRequestEnvelopeError> {
-  const logger = getLogger(config, 'graphql-adapter#rawGraphQLRequestToEnvelope')
-  logger.debug('Received GraphQL request: \n- Headers: ', request.headers, '\n- Body: ', request.body)
-  const requestID = UUID.generate() // TODO: Retrieve request ID from request
-  const eventType = 'MESSAGE' // TODO: (request.requestContext?.eventType as GraphQLRequestEnvelope['eventType']) ?? 'MESSAGE',
-  const connectionID = undefined // TODO: Retrieve connectionId if available,
+  const requestID = UUID.generate()
+  return isExpressWebSocketMessage(request)
+    ? expressWebSocketMessageToEnvelope(config, request, requestID)
+    : expressHttpMessageToEnvelope(config, request, requestID)
+}
+
+function expressWebSocketMessageToEnvelope(
+  config: BoosterConfig,
+  webSocketRequest: ExpressWebSocketMessage,
+  requestId: UUID
+): GraphQLRequestEnvelope | GraphQLRequestEnvelopeError {
+  const logger = getLogger(config, 'graphql-adapter#expressWebSocketMessageToEnvelope')
+  logger.debug('Received  WebSocket GraphQL request: ', webSocketRequest)
+
+  let eventType: EventType = 'MESSAGE'
+  const incomingMessage = webSocketRequest.incomingMessage
+  const headers = incomingMessage?.headers
+  const data = webSocketRequest.data as GraphQLOperation | GraphQLClientMessage | undefined
   try {
+    const connectionContext = webSocketRequest.connectionContext
+    eventType = connectionContext?.eventType
     return {
-      requestID,
-      eventType,
-      connectionID,
-      token: request.headers.authorization,
-      value: request.body,
+      requestID: requestId,
+      eventType: eventType,
```

Comment:
You can actually rename `requestId` to `requestID` and make this like:
```suggestion
      requestID,
      eventType,
```